### ISSUE TITLE
Add Average Precision and R2 to the metrics

### ISSUE THREAD

**Issue by KumoLiu** [Project-MONAI/MONAI#8085]
### Discussed in https://github.com/Project-MONAI/MONAI/discussions/8081

<div type='discussions-op-text'>

<sup>Originally posted by **thibaultdvx** September 13, 2024</sup>
Hi, 
I would like to suggest to add the [**Average Precision**](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html) in the classification metrics. I think that the implementation would be very similar to ROC AUC, so it wouldn't be a big deal to integrate it.
I would be happy to make a PR to do so.

I also feel that an important regression metric is missing:[ **R2 score**](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html). Is there any reason why?
Once again, I would be happy to help to implement it!

WDYT?</div>
